File: 0387-first-unique-character-in-a-String/leetcode_approach.py
# Time complexity: O(N), Space complexity: O(1)
class Solution:
    def firstUniqChar(self, s: str) -> int:

        counts = {}
        for ch in s:
            counts[ch] = 0
        
        for ch in s:
            counts[ch]+= 1

        size = len(s)
        for i in range(size):
            if counts[s[i]] == 1:
                return i

        return -1

File: 0387-first-unique-character-in-a-String/leetcode_approach.cpp
// Time complexity: O(n), Space complexity: O(1)
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> counts;
        for (char ch : s) {
            counts[ch]++;
        }
        int size = s.length();
        for (int i = 0; i < size; i++) {
            char ch = s[i];
            if (counts[ch] == 1) {
                return i;
            }
        }

        return -1;
    }
};